_test/test_algorithms/test_cut.m

```diff
@@ -1,273 +1,292 @@
-        sqw_obj = sqw(obj.sqw_file);
-
-        outfile = fullfile(tmp_dir, 'tmp_outfile.sqw');
-
-        cut(sqw_obj, obj.ref_params{:}, outfile);
-        cleanup = onCleanup(@() clean_up_file(outfile));
-
-        loaded_sqw = sqw(outfile);
-        ref_sqw = sqw(obj.ref_file);
-        assertEqualToTol(loaded_sqw, ref_sqw, obj.FLOAT_TOL, 'ignore_str', true);
-    end
-
-    function test_you_can_take_a_cut_from_a_dnd_object(obj)
-        dnd_obj = d4d(obj.sqw_file);
-
-        u_axis_lims = [-0.1, 0.024, 0.1];
-        v_axis_lims = [-0.1, 0.024, 0.1];
-        w_axis_lims = [-0.1, 0.1];
-        en_axis_lims = [105, 1, 114];
-
-        res = cut(dnd_obj, u_axis_lims, v_axis_lims, w_axis_lims, en_axis_lims);
-        assertTrue(isa(res, 'd3d'));
-
-        % We expect 3 dimensions since we are integrating over w (u3), as
-        % numel(w_axis_lims) = 2.
-        % We expect 9 in the u dimension because the range defined in
-        % u_axis_lims has 9 steps - you can justify this to yourself by
-        % evaluating `numel(u_axis_lims(1):u_axis_lims(2):u_axis_lims(3))`.
-        % For similar reasons, v and en have 9 and 10 dims respectively.
-        expected_img_size = [9, 9, 10];
-        assertEqual(size(res.s), expected_img_size);
-    end
-
-    function test_you_can_take_multiple_cuts_over_integration_axis(obj)
-        proj = projaxes([1, -1 ,0], [1, 1, 0], 'uoffset', [1, 1, 0], 'type', 'paa');
-
-        u_axis_lims = [-0.1, 0.025, 0.1];
-        v_axis_lims = [-0.1, 0.025, 0.1];
-        w_axis_lims = [-0.1, 0.1];
-
-        % Short-hand for defining multiple integration ranges (as opposed to a loop).
-        en_axis_lims = [106, 4, 114, 4];
-        % The indices are as follows:
-        %   1 - first range center
-        %   2 - distance between range centers
-        %   3 - final range center
-        %   4 - range width
-        % Hence the above limits define three cuts, each cut integrating over a
-        % different energy range. The first range being 104-108, the second
-        % 108-112 and the third 112-116.
-
-        sqw_obj = sqw(obj.sqw_file);
-        res = cut(...
-            sqw_obj, proj, u_axis_lims, v_axis_lims, w_axis_lims, en_axis_lims ...
-        );
-
-        expected_en_int_lims = {[104, 108], [108, 112], [112, 116]};
-
-        assertTrue(isa(res, 'sqw'));
-        assertEqual(size(res), [3, 1]);
-        for i = 1:numel(res)
-            assertEqual(size(res(i).data.s), [9, 9]);
-            assertEqual(res(i).data.iint(3:4), expected_en_int_lims{i});
+    
+    methods
+        
+        function obj = test_cut(varargin)
+            save_reference = false;
+            if nargin == 0
+                name = 'test_cut';
+            else
+                if strcmpi(varargin{1},'-save')
+                    save_reference = true;
+                    name = 'test_cut';
+                else
+                    name = varargin{1};
+                end
+            end
```

Comment:
? you mean cellarray as input ?  Well, you run tests with reasonable inputs. If you provided invalid inputs, your rub will probably fail. But you are developer -- I do not believe its difficult for you to see what is wrong to clog the test input with unnecessary checks. 

Just in case, I will add couple of rows describing non-standard test with save class.

Reply:
@abuts The option {name,-save} is not particularly generic. We already have the cases varargin={name} and varargin={-save}. This just combines them - seems quite plausible. Agreed we do not have to concentrate on other generic input.

Reply:
not sure its useful addition, but will add parser for 'some_name','-save' input (who would use this? -- not sure at all)

Reply:
normally custom name may be provided by runtests -- as far as I remember it sometimes make custom names (or was making custom names). The logic may need to be simplified but its very low priority task